weave/Scripts/Main.cs

```diff
@@ -30,50 +29,44 @@ public partial class Main : Node2D
     /// </summary>
     private int _roundCompletions;
 
-    private readonly IList<CurveSpawner> _curveSpawners = new List<CurveSpawner>();
-
     public override void _Ready()
     {
         this.GetNodes();
         if (_keybindings.Count < _nPlayers)
+        {
             throw new ArgumentException(
                 "More players than available keybindings",
                 nameof(_nPlayers)
             );
+        }
 
         SpawnPlayers();
-        InitiateCurveSpawners();
         ClearAndSpawnGoals();
     }
 
     public override void _PhysicsProcess(double delta)
     {
-        CheckPlayerCollisions();
+        DetectPlayerCollision();
     }
 
-    private void CheckPlayerCollisions()
+    private void DetectPlayerCollision()
     {
+        // Collect all segments
+        var allSegments = new HashSet<SegmentShape2D>();
```

Comment:
Again, not needed to be changed, but can be written in one line with linq magic, just fun syntax : -)